### ISSUE TITLE
SC2086: 

### ISSUE THREAD

**Issue by LukeShu** [koalaman/shellcheck#884]
#### For bugs
- Rule Id: 2086
- My shellcheck version: online
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

When encountering an expression of the form `${PARAM:+WORD}`, shellcheck correctly avoids triggering SC2086 on the `${PARAM...}` expression (because it is the `WORD` sub-expression that needs to be quoted), but then fails to enforce SC2086 on the `WORD` sub-expression.

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
ary=(a b c d)
printf '%s\n' ${ary:+${ary[@]}}
```

#### Here's what shellcheck currently says:

```
No issues detected!
```

#### Here's what I wanted or expected to see:

```
Line 3:
printf '%s\n' ${ary:+${ary[@]}}
                     ^-- SC2086: Double quote to prevent globbing and word splitting.
```

I want it to tell me to write it as

```sh
#!/bin/bash
ary=(a b c d)
printf '%s\n' ${ary:+"${ary[@]}"}
```

**Comment by nepella** [comment]
The problem is more obvious with an array having elements that contains spaces:

```bash
#!/bin/bash
ary=(first\ element second third fourth)
printf '%s\n' ${ary:+${ary[@]}}
```

Output:

    first
    element
    second
    third
    fourth

Adding quotes (either around the inner WORD or the entire expression) gives:

```bash
#!/bin/bash
ary=(first\ element second third fourth)
printf '%s\n' ${ary:+"${ary[@]}"}
```

    first element
    second
    third
    fourth
